src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1284 @@
+          if (has(node.parent, 'id') && !isEmpty(node.parent.id)) {
+            parentId = getUuid(node.parent.id)
+            nodeObj.data.parent = parentId
+          }
+          if (has(node, 'runpercent') && !isEmpty(node.runpercent) && (parseInt(node.runpercent) > 0)) {
+            nodeObj.data.runpercent = parseInt(node.runpercent)
+            nodeObj.running = nodeObj.data.runpercent
+            todo = (100 - parseInt(node.runpercent))
+            nodeObj.data.todo = todo
+          }
+          nodesArray.push(nodeObj)
+        })
+        console.log('NODES ::: ', nodesArray)
+        return nodesArray
+      } catch (error) {
+        console.error('getNodes error: ', error)
+      }
+    },
+
+    getEdges (edges) {
+      try {
+        const edgesArray = []
+        let edgeObj = {}
+        each(edges, (edge, key) => {
+          edgeObj = {
+            data: {
+              id: '',
+              source: '',
+              target: '',
+              label: ''
+            },
+            position: {
+
+            },
+            group: 'edges',
+            removed: false,
+            selected: false,
+            selectable: true,
+            locked: false,
+            grabbable: true,
+            classes: ''
+          }
+          has(edge, 'id') && !isEmpty(edge.id) ? edgeObj.data.id = getUuid(edge.id) : console.log('workflowUpdated - edge id is empty')
+          has(edge, 'source') && !isEmpty(edge.source) ? edgeObj.data.source = getUuid(edge.source) : edgeObj.source = undefined
+          has(edge, 'target') && !isEmpty(edge.target) ? edgeObj.data.target = getUuid(edge.target) : edge.target = undefined
+          has(edge, 'label') && !isEmpty(edge.label) ? edgeObj.data.label = edge.label : edgeObj.label = ''
+          edgeObj.data.source !== undefined || edgeObj.data.target !== undefined ? edgesArray.push(edgeObj)
+            : console.log('skipping adding edge with empty source or target')
+        })
+        // console.log('EDGES ::: ', edgesArray)
+        return edgesArray
+      } catch (error) {
+        console.error('getEdges error: ', error)
+      }
+    },
+
+    // for websocket functionality
+    // async messageReceived (msg) {
+    //   console.log('graph view messageRecieved: ', msg)
+    //   this.graphData = JSON.parse(msg.data) // update via watcher
+    // },
+
+    changeLayout (value) {
+      this.layoutName = value
+    },
+
+    async handleMounted () {
+      try {
+        this.status = 'success'
+      } catch (error) {
+        console.error('handleMounted error: ', error)
+        this.status = 'error'
+      }
+    },
+
+    async preConfig (cytoscape) {
+      // cytoscape: this is the cytoscape constructor
+      try {
+        console.log('PRE-CONFIG')
+        cytoscape.use(cola)
+        cytoscape.use(dagre)
+        cytoscape.use(coseBilkent)
+        cytoscape.use(klay)
```

Comment:
Maybe we could remove the different graph types, and use a single style?

This would reduce the code in the PR, making it easier to be reviewed. Then later we can add a PR that brings it back?

Reply:
is this the consensus?

Reply:
Definitely not the consensus, just a suggestion. Someone from @cylc/core may chime in later and give their opinion too. So no need to change it if you don't think we need to do it now :+1:

Reply:
We primarily want dagre (most like cylc-7 graphviz "dot" layout) but need to decide whether or not other layouts would be useful too (for full or partial graphs).  Let's leave it for the moment, I'll try to put some more thought into this PR tomorrow.

Reply:
We probably only want Dagre, however, if there is a semi-usable alternative which can calculate complex layouts faster we may want to auto-switch for large graphs.